packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2273,8 +2273,13 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   @override
   void didChangeMetrics() {
-    if (_lastBottomViewInset < WidgetsBinding.instance!.window.viewInsets.bottom) {
-      _scheduleShowCaretOnScreen();
+    if (_lastBottomViewInset != WidgetsBinding.instance!.window.viewInsets.bottom) {
+      SchedulerBinding.instance!.addPostFrameCallback((Duration _) {
+        _selectionOverlay?.updateForScroll();
+      });
+      if (_lastBottomViewInset < WidgetsBinding.instance!.window.viewInsets.bottom) {
+        _scheduleShowCaretOnScreen();
```

Comment:
I guess this is called only when the bottom inset increases, not decreases, because that's the only way the caret could end up offscreen?  My added `updatedForScroll` call above can be affected by a decrease or increase.  Not sure what the history is on _scheduleShowCaretOnScreen though.

Reply:
iirc @goderbauer added the reveal caret logic. I think at some point I tried to account for phone rotation (portrait -> landscape) but didn't end up doing it for reasons I don't remember.

Reply:
> I guess this is called only when the bottom inset increases, not decreases, because that's the only way the caret could end up offscreen?

Yes, when I looked at it, the only way the caret could end up offscreen was when the keyboard came up and covered the text field.

Reply:
Thanks, that's reassuring!